SpatialGDK/Source/Private/Utils/ComponentReader.cpp

```diff
@@ -103,7 +103,42 @@ void ComponentReader::ApplySchemaObject(Schema_Object* ComponentObject, UObject*
 			{
 				if (Cmd.Type == REPCMD_DynamicArray)
 				{
-					ApplyArray(ComponentObject, FieldId, Cast<UArrayProperty>(Cmd.Property), Data, SwappedCmd.Offset, Cmd.ParentIndex);
+					UArrayProperty* ArrayProperty = Cast<UArrayProperty>(Cmd.Property);
+					bool bProcessedArray = false;
+
+					// Check if this is a FastArraySerializer array so we can simulate the FFastArraySerializerItem PreReplicatedRemove and PostReplicatedAdd calls.
+					if (UStructProperty* ParentStruct = Cast<UStructProperty>(Parent.Property))
+					{
+						if (ParentStruct->Struct->IsChildOf(FFastArraySerializer::StaticStruct()))
```

Comment:
Maybe worth adding a note that this won't handle a case of FastArraySerializer being inside another struct.